src/pages/home/report/ReportActionItemMessageEdit.js

```diff
@@ -161,6 +200,10 @@ class ReportActionItemMessageEdit extends React.Component {
                         selection={this.state.selection}
                         onSelectionChange={this.onSelectionChange}
                     />
+                    <EmojiPicker
+                        isDisabled={isBlockedFromConcierge || isArchivedChatRoom}
+                        onEmojiSelected={this.addEmojiToTextBox}
+                    />
```

Comment:
This can cause performance issues.  Modal is an expensive component and Here it is getting nested into each action item.

Maybe we can try the singleton approach, used for ContextMenu. What do you think?

Reply:
I had the same thought process. Thanks for pointing out ContextMenu. I was looking at `UpdateApp` model component to replicate the behavior.  I'll definitely checkout `ContextMenu`.